pallets/did/src/did_details.rs

```diff
@@ -612,28 +610,36 @@ impl<T: Config> DidDetails<T> {
 	}
 }
 
-pub(crate) type DidNewKeyAgreementKeySet<T> = BoundedBTreeSet<DidEncryptionKey, <T as Config>::MaxNewKeyAgreementKeys>;
-pub(crate) type DidKeyAgreementKeySet<T> = BoundedBTreeSet<KeyIdOf<T>, <T as Config>::MaxTotalKeyAgreementKeys>;
-pub(crate) type DidPublicKeyMap<T> =
-	BoundedBTreeMap<KeyIdOf<T>, DidPublicKeyDetails<BlockNumberOf<T>>, <T as Config>::MaxPublicKeysPerDid>;
+type DidNewKeyAgreementKeySet<MaxNewKeyAgreementKeys> = BoundedBTreeSet<DidEncryptionKey, MaxNewKeyAgreementKeys>;
+pub type DidNewKeyAgreementKeySetOf<T> = DidNewKeyAgreementKeySet<<T as Config>::MaxNewKeyAgreementKeys>;
```

Comment:
Here is a strange bug. If I define the visibility as `pub(crate)` I get a lint warning (dead code) even if the alias is used in the crate (see mock_utils.rs). 

For reference, see: [577d0c58d4](https://github.com/KILTprotocol/kilt-node/pull/530/commits/577d0c58d4755253dbd04e6284eeb8d42862dad1)

Reply:
Yes, because `mock_utils` is only used with the `mock` feature and `test` target. So it should not only be `pub(crate)` but also feature-gated.

Reply:
Ah! Yes – stupid me. :smiley: